third_party/terraform/resources/resource_dataflow_job.go

```diff
@@ -353,3 +457,32 @@ func resourceDataflowJobUpdateJob(config *Config, project string, region string,
 	}
 	return config.clientDataflow.Projects.Locations.Jobs.Update(project, region, id, job).Do()
 }
+
+func resourceDataflowJobLaunchTemplate(config *Config, project string, region string, gcsPath string, request *dataflow.LaunchTemplateParameters) (*dataflow.LaunchTemplateResponse, error) {
+	if region == "" {
+		return config.clientDataflow.Projects.Templates.Launch(project, request).GcsPath(gcsPath).Do()
+	}
+	return config.clientDataflow.Projects.Locations.Templates.Launch(project, region, request).GcsPath(gcsPath).Do()
+}
+
+func resourceDataflowJobReadTemplateStorageObjectHash(gcsPath string, config *Config) (string, error) {
+	// ValidateFunc ensures template_gcs_path matches "gs://<bucket>/<object_path>"
+	parts := strings.SplitN(gcsPath, "/", 4)
+	sObject, err := config.clientStorage.Objects.Get(parts[2], parts[3]).Projection("full").Do()
```

Comment:
Since we're adding a new API call here, it's probably worth mentioning in the changelog that dataflow now requires storage object read permissions